lib/customer-deposit-wallet/test/scenario/Test/Scenario/Wallet/Deposit/Exchanges.lhs.md

```diff
@@ -0,0 +1,218 @@
+
+import Cardano.Crypto.Wallet
+    ( XPrv
+    , XPub
+    )
+import Cardano.Wallet.Deposit.IO
+    ( WalletEnv
+    , WalletInstance
+    )
+import Cardano.Wallet.Deposit.Pure
+    ( Customer
+    , TxSummary (..)
+    , ValueTransfer (..)
+    )
+import Cardano.Wallet.Deposit.Read
+    ( Address
+    , Value
+    )
+import Test.Scenario.Blockchain
+    ( ScenarioEnv
+    , ada
+    , assert
+    , payFromFaucet
+    , signTx
+    , submitTx
+    )
+
+import qualified Cardano.Wallet.Deposit.IO as Wallet
+import qualified Data.Map as Map
+```
+
+We use a function `depositFundsAt` to model the act where a customer deposits funds at a given address.
+
+```haskell
+depositFundsAt :: ScenarioEnv -> Address -> Value -> IO ()
+depositFundsAt env address value = payFromFaucet env [(address, value)]
+```
+
+## 0. Start a Wallet
+
+A `WalletInstance` denotes a mutable wallet that is actively synchronizing to the blockchain, continuously writes its state to a database file, and responds to queries.
+
+In order to create a fresh wallet, or in order to restore a wallet from its public key all the way from genesis, use the function `withWalletInit`. In addition to the public key, this function expects a number which equals the numerically largest customer ID previously handled with this wallet.
+
+```haskell
+scenarioRestore
+    :: XPub -> WalletEnv IO -> IO ()
+scenarioRestore xpub env = do
+    let knownCustomerCount = 127
+    Wallet.withWalletInit env xpub knownCustomerCount $ \w -> do
+        value <- Wallet.availableBalance w
+        assert $ value == ada 0
+```
+
+In order to load the wallet state from a database file and resume operation from that state use the function `withWalletLoad`.
+
+```haskell
+scenarioStart
+    :: WalletEnv IO -> IO ()
+scenarioStart env =
+    Wallet.withWalletLoad env $ \w -> do
+        value <- Wallet.availableBalance w
+        assert $ value == ada 0
+```
+
+## 1. Assign an address to a customer ID
+
+The type `Customer` represents a numerical customer ID.
+Given such a customer ID, the function `createAddress` will create an address and add the association between the customer and this address to the wallet state.
+
+(The mapping from customer ID to address is deterministic and based on the [BIP-32][] address derivation scheme.)
+
+  [bip-32]: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+
+
+The function `listCustomers` returns the associations between customers and addresses recorded in the current wallet state.
+
+```haskell
+scenarioCreateAddressList
+    :: WalletInstance -> IO ()
+scenarioCreateAddressList w = do
+    let customer = 31
+    address <- Wallet.createAddress customer w
+    customers <- Wallet.listCustomers w
+    assert $ (customer, address) `elem` customers
+```
+
+## 2. Track deposits at this address
+
+As soon as an association between customer and address has been added to the wallet state using `createAddress`, the wallet will track deposits made at this address.
```

Comment:
```suggestion
As soon as an association between customer and address has been established to the wallet state using `createAddress`, the wallet will track deposits sent to this address.
```

Reply:
I prefer "added to the wallet state" over "established in the wallet state",
but I like "made at" less than "sent to".